### PR TITLE
bt-61: create overlay component | development

### LINKED ISSUE #61: Create Overlay component
It should be a component that shows screen data but disables all possible touchable items. It also should have a spinner in the center. Also, this component should be fullscreen and almost transparent so that the user can see the screen content.

Example of usage: 
1. User makes some async action (login for example)
2. While waiting for a response user sees Overlay with a spinner and cannot press buttons

### PR DESCRIPTION
**Pull request by mskmee**
image

Made misstake and create branch without task prefix, can't delete it (

### CODE REVIEW

File: mobile/src/bundles/common/components/overlay/overlay.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    color?: string;
+};
+const Overlay: React.FC<OverlayProperties> = ({
+    isActive,
+    color = Color.PRIMARY,
+}) => {
+    if (!isActive) {
+        return null;
+    }
+    return (
```
Comment (likeri29) [inline mobile/src/bundles/common/components/overlay/overlay.tsx:23]: I think better approach is:
`return ({isActive ? <View
            style={[
                globalStyles.width100,
                globalStyles.height100,
                globalStyles.justifyContentCenter,
                styles.wrapper,
            ]}
        >
            <ActivityIndicator color={color} size="large" />
        </View> : null})`

Comment (mskmee) [reply mobile/src/bundles/common/components/overlay/overlay.tsx:23]: Always write like this, if we will write in syntax like you write, I'll change component

Comment (nikita-remeslov) [reply mobile/src/bundles/common/components/overlay/overlay.tsx:23]: @likeri29 syntax like this more readable, as for me
so I think it's best to stick with it!

@mskmee I am with you 👍

**Review by nikita-remeslov — APPROVED** [review]

File: mobile/src/bundles/common/components/overlay/overlay.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        <View
+            style={[
+                globalStyles.width100,
+                globalStyles.height100,
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/overlay/overlay.tsx:28]:
```suggestion
                globalStyles.flex1,
```

File: mobile/src/bundles/common/components/overlay/overlay.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+};
+const Overlay: React.FC<OverlayProperties> = ({
+    isActive,
+    color = Color.PRIMARY,
+}) => {
+    if (!isActive) {
+        return null;
+    }
+    return (
+        <View
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/overlay/overlay.tsx:24]:
```suggestion
        <View
        pointerEvents="box-only"
```

File: mobile/src/bundles/common/components/overlay/overlay.tsx (hunk 2 above)
Comment (mskmee) [reply mobile/src/bundles/common/components/overlay/overlay.tsx:28]: Flex: 1 didnt resize absolute component, or i should use it in another way?

File: mobile/src/bundles/common/components/overlay/overlay.tsx (hunk 3 above)
Comment (mskmee) [reply mobile/src/bundles/common/components/overlay/overlay.tsx:24]: fix

**Review by katerynakostikova — APPROVED** [review]

**Pull request merged**